FinancialTransactionsApi/V1/Gateways/DynamoDbGateway.cs

```diff
@@ -118,5 +130,88 @@ public async Task UpdateAsync(Transaction transaction)
         {
             await _dynamoDbContext.SaveAsync(transaction.ToDatabase()).ConfigureAwait(false);
         }
+
+        public async Task<List<Transaction>> GetAllTransactionsForTheYearAsync(ExportTransactionQuery query)
```

Comment:
Can this endpoint be made generic please? Possibly consider combining with GetAllTransactionsAsync, where start/end date are optional filters. Current method name implies that the only supported filter is transactions for last year, instead of for any start/end date range.

Reply:
Code refactored